bundler/lib/bundler/compact_index_client/parser.rb

```diff
@@ -22,6 +22,7 @@ def versions
 
         lines(@compact_index.versions).each do |line|
           name, versions_string, checksum = line.split(" ", 3)
+          next unless versions_string # Artifactory has a bug where blank lines exist in the versions file
```

Comment:
I reluctantly added this, even though there's no previous expectation that it should handle blank lines in the file.